### PR TITLE
[chip-tool] Add commands to generate/update onboarding payloads

### PR DESCRIPTION
**Pull request by vivien-apple**
#### Problem

`chip-tool` is not able to generate onboarding payloads. Furthermore there is no way to generate invalid payloads using the current APIs.

Partially fix #16385

#### Change overview
  * Add a command to `chip-tool` to generate/update a qr code

  * Add a command to `chip-tool` to generate/update a manual code

  *Update the APIs provided by `ManualSetupPayloadGenerator` and `QRCodeSetupPayloadGenerator` to allow for creating invalid payloads. 
 
#### Testing
It was tested locally and it will be used to generate payloads for TC-DD-3.12/13

### CODE REVIEW

File: examples/chip-tool/commands/payload/SetupPayloadGenerateCommand.cpp [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+    ChipLogProgress(chipTool, "QR Code: %s", code.c_str());
+
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR SetupPayloadGenerateManualCodeCommand::Run()
+{
+    SetupPayload payload;
+
+    if (mPayload.HasValue())
```
Comment (bzbarsky-apple) [inline examples/chip-tool/commands/payload/SetupPayloadGenerateCommand.cpp:86]: Could we factor the duplicated "set all the things on `payload`" code out into a helper method?

File: src/setup_payload/ManualSetupPayloadGenerator.h [hunk 2]
```diff
@@ -70,6 +70,25 @@ class ManualSetupPayloadGenerator
     // Populates decimal string representation of the payload into outDecimalString.
     // Wrapper for using std::string.
     CHIP_ERROR payloadDecimalStringRepresentation(std::string & outDecimalString);
+
+    /**
+     * This function disable internal checks about the validity of the generated payload.
```
Comment (bzbarsky-apple) [inline src/setup_payload/ManualSetupPayloadGenerator.h:75]:
```suggestion
     * This function disables internal checks about the validity of the generated payload.
```

File: src/setup_payload/ManualSetupPayloadGenerator.h [hunk 3]
```diff
@@ -70,6 +70,25 @@ class ManualSetupPayloadGenerator
...
+     * It allows using the generator to generates invalid payloads.
+     * Defaults is false.
```
Comment (bzbarsky-apple) [inline src/setup_payload/ManualSetupPayloadGenerator.h:77]:
```suggestion
     * It allows using the generator to generate invalid payloads.
     * Default is false.
```

File: src/setup_payload/ManualSetupPayloadGenerator.h [hunk 4]
```diff
@@ -70,6 +70,25 @@ class ManualSetupPayloadGenerator
...
+     */
+    void SetAllowInvalidPayload(bool allow) { mAllowInvalidPayload = allow; }
+
+    /**
+     * This function allow to forced the generation of a short code when the commissioning
+     * flow is not standard by ignoring the vendor id and product id informations but with
+     * the VID/PID present flag set.
+     * Defaults is false.
```
Comment (bzbarsky-apple) [inline src/setup_payload/ManualSetupPayloadGenerator.h:85]:
```suggestion
     * This function allows forcing the generation of a short code when the commissioning
     * flow is not standard by ignoring the vendor id and product id informations but with
     * the VID/PID present flag set.
     * Default is false.
```

**Review by bzbarsky-apple — APPROVED** [review]

File: examples/chip-tool/commands/payload/SetupPayloadGenerateCommand.cpp (hunk 1 above)
Comment (vivien-apple) [reply examples/chip-tool/commands/payload/SetupPayloadGenerateCommand.cpp:86]: I have introduced a base class that host most of the options with a method to set up most of the values.

**Review by woody-apple — APPROVED** [review]

**Pull request merged**